File: backend/src/main/java/dev/saul/descarbonizacion_transporte/modules/workers/dto/output/WorkerOutputDto.java
package dev.saul.descarbonizacion_transporte.modules.workers.dto.output;

import dev.saul.descarbonizacion_transporte.modules.worker_positions.dto.output.WorkerPositionOutputDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WorkerOutputDto {
    private String id;

    private String names;

    private String lastNames;

    private String cid;

    private String phoneNumber;

    private String email;

    private WorkerPositionOutputDto position;
}

File: frontend/src/api/workers/interfaces/output/worker.output.dto.ts
import type WorkerPositionOutputDto from '@/api/worker-positions/interfaces/output/worker-position.output.dto.ts'

export default interface WorkerOutputDto {
  id: string
  names: string
  lastNames: string
  cid: string
  phoneNumber: string
  email: string
  position: WorkerPositionOutputDto
}
